kernel/chown.c

```diff
@@ -121,10 +123,12 @@ long myst_syscall_fchown(int fd, uid_t owner, gid_t group)
     if (fd < 0)
         return -EINVAL;
 
-    if (((owner != (uid_t)-1) &&
-         (myst_valid_uid_against_passwd_file(owner) < 0)) ||
-        ((group != (gid_t)-1) &&
-         (myst_valid_gid_against_group_file(group) < 0)))
+    /* if not root, check target uid/gid validity */
+    if ((thread->euid != 0) &&
+        (((owner != (uid_t)-1) &&
```

Comment:
Please write nested if-statements to make the conditions more clear. Also on Linux I do get errors with `sudo chmod xxxx <file>` when `xxxx` is a string and doesn't map to an existing user/group. It seems Ok when `xxxx` is a number.